### DISCUSSION TITLE
Cypress: Alternative for command 'wait'

### DISCUSSION THREAD

**Discussion by JanniChi** [cypress-io/cypress#19996] [Cypress Cloud]
I am looking for an alternative to the wait command.

As we all know, a page/component/etc. does not take the same time to load or simply be available every time it is called.
For this reason, I cannot rely on the 'wait' command with constant time specification.

So I am looking for an alternative.

Some people on the internet write that the command 'then' helps to wait until the condition is fulfilled.
Unfortunately, the following alternative does not work in my case:

`cy.get(#element).should('exist').then(() =>{
cy.get(#element).click();
});`

I am looking forward to your messages and alternatives to 'wait'.
Many greetings Janni

**Comment by bahmutov** [comment]
This kind of wait (the element should exist) is built into the `cy.get` + `cy.click` commands. For example, `cy.get` by default checks if the element exists https://on.cypress.io/get#Assertions The command `cy.click` goes beyond that - it checks if the element exists, is visible, is actionable, etc. https://on.cypress.io/click#Assertions

So you example is better written as `cy.get('...).click()`. 

Tip: look at the timeouts, you can control how long each commands "waits" internally for the element https://on.cypress.io/retry-ability
Tip 2: even better, read https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/
